tests/modules/shasta/test.yml

```diff
@@ -0,0 +1,45 @@
+- name: shasta test_shasta
+  command: nextflow run ./tests/modules/shasta -entry test_shasta -c ./tests/config/nextflow.config  -c ./tests/modules/shasta/nextflow.config
+  tags:
+    - shasta
+  files:
+    - path: output/shasta/ShastaRun/Assembly-BothStrands-NoSequence.gfa
+      contains: "[ # TODO nf-core: file md5sum was variable, please replace this text with a string found in the file instead ]"
```

Comment:
These need dealing with, it shows that the test generated different output files when it was run twice.
This might just be that the file contains a timestamp, non-deterministic ordering of results, or rounding errors, or it might mean there is a random seed that needs to be set in the program (at which point a default value for the random seed that can be overridden by a config would be the best option).
If it is just a timestamp or similar, then you can just check for a string as the TODO states. Or drop the test check altogether as you have a lot of files being emitted here.

Reply:
I've checked the flagged files with the TO-DO statement, and s not timestamps or such. What actually varies is the assembly itself between runs of the software.

I saw that the Assembly.fasta is there, correct, as expected, is an assembled genome. But the md5sum checks are different because some nucleotides may have changed. And, this, causes all the others to also change as all of the other ones flagged are relative to the Assembly itself.

So, I guess the most relevant for those files would be to check whether they exist only.

Reply:
Is there an option to set the random seed on the tool via the command line?

Reply:
Not that I am aware of. These are all the command line options: https://chanzuckerberg.github.io/shasta/CommandLineOptions.html

And all of them seem to be only defaults that keep the same across executions ... I did not see something like a random seed. :/

Reply:
Well that is both awkward and strange that it isn't included. So running the tool twice gives different actual output because of the randomness? The code of the tool itself seems to set the seed to 231 in [several places]:(https://github.com/chanzuckerberg/shasta/search?l=C%2B%2B&p=1&q=seed), but I don't understand C++ to actually follow the structure of the code.

Reply:
Can you ask on the tool github?

Reply:
Let’s wait:

https://github.com/chanzuckerberg/shasta/issues/296#issue-1390697538

Reply:
Thanks, we'll see what they say.

Reply:
Hmm, ok. Can you make the tests pass. You can check that files contain a string of text if that is possible as the TODO mentions, or you can just require the existence of the file (can't remember the syntax for that off the top of my head).

Reply:
That’s how it is at the moment, you can see that the tests are passing because I changed to only check for existence